libs/agno/agno/tools/openai.py

```diff
@@ -36,3 +66,119 @@ def transcribe_audio(self, audio_path: str) -> str:
+            size (OpenAIImageSize): The size of the generated images. Defaults to "1024x1024".
+                - For dall-e-2: Must be one of "256x256", "512x512", or "1024x1024".
+                - For dall-e-3: Must be one of "1024x1024", "1792x1024", or "1024x1792".
+
+        Returns:
+            str: A comma-separated string of URLs pointing to the generated images, or an error message.
+        """
+        log_info(f"Generating {n} image(s) using {model_name} of size {size} for prompt: '{prompt}'")
+
+        if model_name == "dall-e-3":
+            if n != 1:
+                logger.warning("DALL-E 3 only supports n=1. Forcing n=1.")
+                n = 1  # Force n=1 for DALL-E 3
+            if size not in ["1024x1024", "1792x1024", "1024x1792"]:
+                error_msg = f"Invalid size '{size}' for DALL-E 3. Must be '1024x1024', '1792x1024', or '1024x1792'."
+                logger.error(error_msg)
+                return error_msg
+        elif model_name == "dall-e-2":
+            if size not in ["256x256", "512x512", "1024x1024"]:
+                error_msg = f"Invalid size '{size}' for DALL-E 2. Must be '256x256', '512x512', or '1024x1024'."
+                logger.error(error_msg)
+                return error_msg
+
+        # --- API Call ---
+        try:
+            response = client.images.generate(
+                model=model_name,
+                prompt=prompt,
+                n=n,
+                size=size,
+                response_format="url",
+            )
+            image_urls = [item.url for item in response.data if item.url]
+
+            # Add images to agent context
+            for url in image_urls:
+                if url:  # Ensure URL is not empty
+                    media_id = str(uuid4())
+                    agent.add_image(
+                        ImageArtifact(
+                            id=media_id,
+                            url=url,
+                            prompt=prompt,  # Optionally add prompt context
+                            model=model_name,  # Optionally add model context
+                        )
+                    )
+                    log_info(f"Added generated image artifact {media_id} to agent.")
+
+            result_string = ", ".join(image_urls)
+            log_info(f"Generated image URLs (as string): {result_string}")
+            return result_string
+        except Exception as e:
+            logger.error(f"Failed to generate image using {model_name}: {str(e)}")
+            return f"Failed to generate image using {model_name}: {str(e)}"
+
+    def generate_speech(
+        self,
+        agent: Agent,
+        text_input: str,
+    ) -> str:
+        """Generate speech from text using OpenAI's Text-to-Speech API.
+        Adds the generated audio as an AudioArtifact to the agent's context.
+        Uses the voice, model, and format specified during OpenAITools initialization.
```

Comment:
This function is for the model. This information is not useful to the model and we should remove it.